src/providers/openai.ts

```diff
@@ -30,6 +30,8 @@ type OpenAiCompletionOptions = OpenAiSharedOptions & {
   best_of?: number;
   functions?: OpenAiFunction[];
   function_call?: 'none' | 'auto' | { name: string };
+  tools?: OpenAiTool[];
+  tool_choice?: 'none' | 'auto' | { name: string };
```

Comment:
According to the [OpenAI docs](https://platform.openai.com/docs/api-reference/chat/create#chat-create-tools), should this be
```ts
{
  // ...
  tools_choice?: 'none' | 'auto' | { type: 'function'; function?: { name: string }}
}
```

Reply:
done